### DISCUSSION TITLE
How to utilize the metric message

### DISCUSSION THREAD

**Discussion by lingyan98** [srsran/srsRAN_Project#940] [General Help]
Hello everyone.

I'm deploying srsRAN+flexric. I have tested the xapp_oran_moni.py and the KPM message can be shown in my terminal, shown as follows:
![image](https://github.com/user-attachments/assets/3b68615b-62e9-41e9-b2cd-46bbd0109e52)

### Questions
I'm wondering how I can get these values (like saving them into a JSON file or getting them as a Python value) in my code. Are there any tutorials?

**Comment by pgawlowicz** [comment]
The values are already in Python, as you get them printed in `xapp_oran_moni.py` (Flexric xApp).

**Reply by lingyan98** [reply]
> The values are already in Python, as you get them printed in `xapp_oran_moni.py` (Flexric xApp).

Thank you for your reply. These values have been printed in the terminal. But I want to use them to return them as Python values. The following image shows the code in the "xapp_oran_moni.py". The function "report_kpm_sm" returns a hndlr. If I want to return the KPM, Do I need to modify this function in the C++ file?
![image](https://github.com/user-attachments/assets/5cc824a9-91b2-4e18-99fe-91a5189ce4dd)

**Reply by pgawlowicz** [reply]
Probably they are printed in the `KPMCallback` object, that is also defined in the same file.

**Reply by lingyan98** [reply]
> Probably they are printed in the `KPMCallback` object, that is also defined in the same file.

Thank you for your replay, I will test this!
